src/nix/path-info.cc

```diff
@@ -86,14 +94,58 @@ struct CmdPathInfo : StorePathsCommand, MixJSON
         for (auto & storePath : storePaths)
             pathLen = std::max(pathLen, store->printStorePath(storePath).size());
 
+        StorePathSet substitutablePaths;
+        if (showSubStatus) {
+            StringSet pathsS;
+            // query each configured substituter
```

Comment:
This should use `Store::querySubstitutablePaths()`.

Reply:
So I tried overriding it but its unreliable it seems unless I override both the `RemoteStore` and `LocalStore` implementations. Also it starts a threadpool in the amount of `fileTransferSettings.httpConnections` and then immediate shuts it down before even starting the queries, not sure why that is as I didn't do that in my override code.